### PR TITLE
DataStore modelgen add support for model-less types

### PR DESCRIPTION
**Pull request by yuth**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-codegen-appsync-model-plugin/src/visitors/appsync-swift-visitor.ts [hunk 1]
```diff
@@ -209,12 +243,18 @@ export class AppSyncSwiftVisitor extends AppSyncModelVisitor {
...
     if (field.isList) {
-      ofType = `.collection(of: ${this.getSwiftModelTypeName(field)})`;
+      if (isModelType) {
+        ofType = `.collection(of: ${this.getSwiftModelTypeName(field)})`;
+      } else {
+        ofType = `.customType(${this.getSwiftModelTypeName(field)})`;
+      }
     } else {
       if (isEnumType) {
         ofType = `.enum(${typeName})`;
```
Comment (drochetti) [inline packages/amplify-codegen-appsync-model-plugin/src/visitors/appsync-swift-visitor.ts]: This should be `.enum(type: ${typeName})`

File: packages/amplify-codegen-appsync-model-plugin/src/visitors/appsync-swift-visitor.ts [hunk 2]
```diff
@@ -75,6 +87,27 @@ export class AppSyncSwiftVisitor extends AppSyncModelVisitor {
     });
     return result.join('\n');
   }
+  generateNonModelType(): string {
+    let result: string[] = [...this.imports, ''];
+    Object.entries(this.getSelectedNonModels()).forEach(([name, obj]) => {
+      const structBlock: SwiftDeclarationBlock = new SwiftDeclarationBlock()
+        .withName(this.getModelName(obj))
```
Comment (drochetti) [inline packages/amplify-codegen-appsync-model-plugin/src/visitors/appsync-swift-visitor.ts]: need to add the `public` access modifier before the `struct` declaration

File: packages/amplify-codegen-appsync-model-plugin/src/visitors/appsync-typescript-visitor.ts [hunk 3]
```diff
@@ -41,15 +41,21 @@ export class AppSyncModelTypeScriptVisitor<
...
+      .map(typeObj => this.generateModelDeclaration(typeObj))
+      .join('\n\n');
 
-    const modelExports = this.generateExports(Object.values(this.typeMap));
+    const modelInitialization = this.generateModelInitialization([...Object.values(this.modelMap), ...Object.values(this.nonModelMap)]);
 
-    return [imports, '', enumDeclarations, '', modelDeclarations, '', modelInitialization, '', modelExports].join('\n');
+    const modelExports = this.generateExports(Object.values(this.modelMap));
+
+    return [imports, '', enumDeclarations, '', modelDeclarations, '', nonModelDeclarations, '', modelInitialization, '', modelExports].join(
```
Comment (attilah) [inline packages/amplify-codegen-appsync-model-plugin/src/visitors/appsync-typescript-visitor.ts]: you could use `'\n\n'` for join for new lines.

File: packages/amplify-codegen-appsync-model-plugin/src/__tests__/visitors/appsync-json-metadata-visitor.test.ts [hunk 4]
```diff
@@ -0,0 +1,510 @@
...
+    it('should return NonModel type for Non-model', () => {
+      expect((visitor as any).getType('SimpleNonModelType')).toEqual({ nonModel: 'SimpleNonModelType' });
+    });
+
+    it('should throw error for unknown type', () => {
+      expect(() => (visitor as any).getType('unknown')).toThrowError('Unknown type');
+    });
+  });
+
+  describe('generateNonModelMetaData', () => {
```
Comment (attilah) [inline packages/amplify-codegen-appsync-model-plugin/src/__tests__/visitors/appsync-json-metadata-visitor.test.ts]: change MetaData -> Metadata everywhere

File: packages/amplify-codegen-appsync-model-plugin/src/visitors/appsync-javascript-visitor.ts [hunk 5]
```diff
@@ -52,20 +52,31 @@ export class AppSyncModelJavascriptVisitor<
...
+      const modelDeclarations = Object.values(this.modelMap)
         .map(typeObj => this.generateModelDeclaration(typeObj, true))
         .join('\n\n');
 
-      return [imports, '', enumDeclarations, '', modelDeclarations].join('\n');
+      const nonModelDeclarations = Object.values(this.nonModelMap)
+        .map(typeObj => this.generateModelDeclaration(typeObj, true))
+        .join('\n\n');
+
+      return [imports, '', enumDeclarations, '', nonModelDeclarations, '', modelDeclarations].join('\n');
```
Comment (attilah) [inline packages/amplify-codegen-appsync-model-plugin/src/visitors/appsync-javascript-visitor.ts]: Same like above for join with empty lines

**Pull request merged**